storey/flow.py

```diff
@@ -206,6 +206,18 @@ def _check_stage_in_flow(self, type_to_check):
                 return True
             if step._check_stage_in_flow(type_to_check):
                 return True
+        if self._recovery_step is not None:
+            if isinstance(self._recovery_step, type_to_check):
+                return True
+            if isinstance(self._recovery_step, dict):
+                for step in self._recovery_step.values():
+                    if isinstance(step, type_to_check):
+                        return True
+                    if step._check_stage_in_flow(type_to_check):
+                        return True
+            else:
+                if self._recovery_step._check_stage_in_flow(type_to_check):
+                    return True
```

Comment:
I think this impl would be a bit more straightforward. Note that we don't use the term "stage" in storey.
```
    def _check_step_in_flow(self, type_to_check):
        if isinstance(self, type_to_check):
            return True
        for outlet in self._outlets:
            if outlet._check_step_in_flow(type_to_check):
                return True
        if isinstance(self._recovery_step, Flow):
            if self._recovery_step._check_step_in_flow(type_to_check):
                return True
        elif isinstance(self._recovery_step, dict):
            for step in self._recovery_step.values():
                if step._check_step_in_flow(type_to_check):
                    return True
        return False
```